### PR TITLE
Create some integration tests for the various B3 propagator options.

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      try (Response r = call.execute()) {
+        ResponseBody body = r.body();
+        assertThat(body.string()).isEqualTo("OK");
+      }
+    } finally {
+      span.end();
+    }
+
+    List<SpanData> finishedSpanItems = spanExporter.getFinishedSpanItems();
+    assertThat(finishedSpanItems).hasSize(3);
```
Comment (jamesylgan) [inline integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:93]: Using a static int variable (with a descriptive name) might make more sense here - otherwise it is unclear why # span items should be `3`

**Review by jamesylgan — APPROVED** [review]
These ITs look great!

Small suggestion - it would be great if these ITs could leverage the classes:
`SdkTracerProviderConfigurer` and `ConfigurablePropagatorProvider` (so we can detect if the system of loading in custom code has changed)

File: integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java (hunk 1 above)
Comment (jkwatson) [reply integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:93]: I can add a comment. I don't think having a static variable would help with clarity here.

Comment (jkwatson) [reply integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:94]: comments added. good suggestion

File: integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+import spark.Service;
+
+/** Integration tests for the B3 propagators, in various configurations. */
+public class B3PropagationIntegrationTest {
+
+  private static final int server1Port = new Random().nextInt(40000) + 1024;
+  private static final int server2Port = new Random().nextInt(40000) + 1024;
+  private static final InMemorySpanExporter spanExporter = InMemorySpanExporter.create();
+
+  private OpenTelemetrySdk clientSdk;
```
Comment (anuraaga) [inline integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:44]: Maybe we don't need these fields (just came to mind since so many SDKs here is a bit hard to follow)

**Review by anuraaga — APPROVED** [review]

File: integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      server2.stop();
+    }
+    if (server1 != null) {
+      server1.stop();
+    }
+    spanExporter.reset();
+  }
+
+  @ParameterizedTest
+  @MethodSource("propagators")
```
Comment (anuraaga) [inline integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:73]: BTW personally prefer ArgumentSource for compile safety and to avoid the suppresswarnings

**Comment by jkwatson** [comment]
> These ITs look great!
> 
> Small suggestion - it would be great if these ITs could leverage the classes:
> `SdkTracerProviderConfigurer` and `ConfigurablePropagatorProvider` (so we can detect if the system of loading in custom code has changed)

I don't understand how adding these components to these tests would help you.

File: integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java (hunk 3 above)
Comment (jkwatson) [reply integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:73]: :thu

File: integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java (hunk 2 above)
Comment (jkwatson) [reply integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java:44]: good idea. I can streamline this a bit.
